### PR TITLE
Added more OUIA props into Dropdown component

### LINKED ISSUE #4753: Better OUIA support in Dropdown
**Describe the issue. What is the expected and unexpected behavior?**

Dropdown subcomponents such as DropoownItem, DropdownToggle and DropdownToggleCheckbox should have support OUIA.

### PR DESCRIPTION
**Pull request by quarckster**
**What**: Closes #4753

This PR adds OUIA support for DropdownItem, DropdownToggle and DropdownToggleCheckbox.

### CODE REVIEW

**Comment by quarckster** [comment]
@redallen @jschuler please review

**Review by redallen — DISMISSED** [review]
LGTM.

Conflicts with #4746, though. cc @jschuler

File: packages/react-core/src/components/Dropdown/DropdownItem.tsx [hunk 1]
```diff
@@ -89,6 +92,7 @@ export const DropdownItem: React.FunctionComponent<DropdownItemProps> = ({
         autoFocus={autoFocus}
         styleChildren={styleChildren}
         description={description}
+        {...getOUIAProps(DropdownItem.displayName, ouiaId, ouiaSafe)}
```
Comment (jschuler) [inline packages/react-core/src/components/Dropdown/DropdownItem.tsx:95]: for better default ids you can follow this comment https://github.com/patternfly/patternfly-react/pull/4769#discussion_r488157601

File: packages/react-core/src/components/Dropdown/DropdownSeparator.tsx [hunk 2]
```diff
@@ -24,6 +27,7 @@ export const DropdownSeparator: React.FunctionComponent<SeparatorProps> = ({
         component={<Divider component={DividerVariant.div} />}
         className={className}
         role="separator"
+        {...getOUIAProps(DropdownSeparator.displayName, ouiaId, ouiaSafe)}
```
Comment (jschuler) [inline packages/react-core/src/components/Dropdown/DropdownSeparator.tsx:30]: if you want better default ids
https://github.com/patternfly/patternfly-react/pull/4769#discussion_r488157601

File: packages/react-core/src/components/Dropdown/DropdownToggleCheckbox.tsx [hunk 3]
```diff
@@ -61,6 +74,7 @@ export class DropdownToggleCheckbox extends React.Component<DropdownToggleCheckb
           aria-invalid={!isValid}
           disabled={isDisabled}
           checked={this.calculateChecked()}
+          {...getOUIAProps(DropdownToggleCheckbox.displayName, ouiaId, ouiaSafe)}
```
Comment (jschuler) [inline packages/react-core/src/components/Dropdown/DropdownToggleCheckbox.tsx:77]: if you'd like better default id here also, for class based components the approach would be to add a state variable
```
import { getOUIAProps, OUIAProps, getDefaultOUIAId } from '../../helpers';
...
export class DropdownToggleCheckbox extends React.Component<DropdownToggleCheckboxProps, { ouiaStateId: string }> {
this.state = {
      ouiaStateId: getDefaultOUIAId(DropdownToggleCheckbox.displayName)
    };
}

return (
  <input {...getOUIAProps(DropdownToggleCheckbox.displayName, ouiaId !== undefined ? ouiaId : this.state.ouiaStateId, ouiaSafe)} />
)
```

**Review by jschuler — CHANGES_REQUESTED** [review]

File: packages/react-core/src/components/Dropdown/DropdownToggleCheckbox.tsx (hunk 3 above)
Comment (quarckster) [reply packages/react-core/src/components/Dropdown/DropdownToggleCheckbox.tsx]: Done

File: packages/react-core/src/components/Dropdown/DropdownItem.tsx (hunk 1 above)
Comment (quarckster) [reply packages/react-core/src/components/Dropdown/DropdownItem.tsx:95]: I'm afraid my poor knowledge of TS is not enough to apply the same trick as in  https://github.com/patternfly/patternfly-react/pull/4769#discussion_r488157601. Where should I put `const ouiaProps = useOUIAProps(DropdownItem.displayName, ouiaId, ouiaSafe);`?

Comment (redallen) [reply packages/react-core/src/components/Dropdown/DropdownItem.tsx:95]: You should change the signature from:
```tsx
export const DropdownItem = ({
  ...
}) => (
  <element />
)
```
to
```tsx
export const DropdownItem = ({
  ...
}) => {
  const ouiaProps = useOUIAProps(NavItem.displayName, ouiaId, ouiaSafe);
  return (
    <element {...ouiaProps} />
  )
}
```

**Review by redallen — CHANGES_REQUESTED** [review]
This looks close, but please use the new hook `useOUIAProps` instead! It fixes problems with the default ids which should make defaults friendlier for testing.

File: packages/react-core/src/components/Dropdown/DropdownItem.tsx (hunk 1 above)
Comment (quarckster) [reply packages/react-core/src/components/Dropdown/DropdownItem.tsx]: @redallen thank you very much!

**Review by jschuler — APPROVED** [review]

**Review by redallen — APPROVED** [review]
LGTM!

**Pull request merged**